controllers/infrastructure/byomachine_controller.go

```diff
@@ -260,19 +260,20 @@ func (r *ByoMachineReconciler) reconcileNormal(ctx context.Context, machineScope
 
 	if machineScope.ByoMachine.Spec.ProviderID == "" {
 		logger.Info("Updating Node with ProviderID")
-		providerID := fmt.Sprintf("%s%s/%s", providerIDPrefix, machineScope.ByoHost.Name, util.RandomString(providerIDSuffixLength))
+
 		remoteClient, err := r.getRemoteClient(ctx, machineScope.ByoMachine)
 		if err != nil {
 			logger.Error(err, "failed to get remote client")
 			return ctrl.Result{}, err
 		}
 
-		err = r.setNodeProviderID(ctx, remoteClient, machineScope.ByoHost, providerID)
+		providerID, err := r.setNodeProviderID(ctx, remoteClient, machineScope.ByoHost, machineScope.ByoHost.Name)
```

Comment:
You are already passing `machineScope.ByoHost`, no need to pass `machineScope.ByoHost.Name`

Reply:
Will fix it.